scipy/integrate/_ivp/ivp.py

```diff
@@ -425,25 +425,25 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
             ts.append(t)
             ys.append(y)
         else:
-            t_step = []
-
-            while t_eval_i < n_eval and s * (t_eval[t_eval_i] - t) < 0:
-                t_step.append(t_eval[t_eval_i])
-                t_eval_i += 1
+            t_end = t_eval_i
+            while t_end < n_eval and s * (t_eval[t_end] - t) < 0:
+                t_end += 1
```

Comment:
Line_profiler reports
```
Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
...
   428        34           58      1.7      0.1              t_end = t_eval_i
   429     20032        36826      1.8     49.7              while t_end < n_eval and s * (t_eval[t_end] - t) < 0:
   430     19998        26925      1.3     36.3                  t_end += 1
   431                                           
```
for trivial problem with `len(t_eval) == 20000`. I'd use searchsorted here to get binary search instead of linear search.